cpp/src/arrow/ipc/reader.cc

```diff
@@ -58,6 +58,8 @@
 #include "generated/Schema_generated.h"
 #include "generated/SparseTensor_generated.h"
 
+#define FLATBUFFER_MAX_TABLES 10000000
```

Comment:
style nit: use a constexpr in an anonymous namespace.

Reply:
or simple comment the parameter inline below /*flatbuffer_max_tables=*/100000000

Reply:
> According to various sources on the Internet it should be ok to just pass `UINT_MAX`. @wesm @emkornfield What do you think?

That should be fine indeed. I got the solution for this problem from:
https://groups.google.com/g/flatbuffers/c/JtDGnBPx9is

> You simply need to pass your own max_tables to the FlatBufferBuilder constructor.
> Pass e.g. INT_MAX and you will guaranteed never hit it :)